doc/releases/changelog-dev.md

```diff
@@ -361,8 +363,15 @@
   Please provide a seed to each individual function instead if you want controllable results.
   [(#3624)](https://github.com/PennyLaneAI/pennylane/pull/3624)
 
+* `qml.dot` no longer corresponds to the function in `pennylane/collections/dot.py`. It now
+  redirects to the function `pennylane/ops/functions/dot.py` instead.
+  [(#3686)](https://github.com/PennyLaneAI/pennylane/pull/3686)
+
```

Comment:
Do we need to go through a deprecation cycle. For example, `qml.dot` dispatches to `pennylane/collections/dot.py` if input arguments are of the right type, *but* raises a deprecation warning when doing so.

Otherwise, `qml.dot` dispatches to our new dot function.

Reply:
Good point! We can definitely do that 👍